pkg/k8shandler/cluster.go

```diff
@@ -50,6 +50,15 @@ func CreateOrUpdateElasticsearchCluster(cluster *api.Elasticsearch) error {
 		clusterStatus := cluster.Status.DeepCopy()
 		index, nodeStatus := getNodeStatus(upgradeInProgressNode.name(), clusterStatus)
 
+		isStuck, err := upgradeInProgressNode.updateStuckPod(nodeStatus)
+		if err != nil {
+			return err
+		}
+		if isStuck {
```

Comment:
I think we want to just iterate over any pods that are stuck based on the `cluster.Status` node conditions and do this before we worry about any upgrades in progress (a stuck node means we wouldn't get around to anything else).